packages/itwinui-react/src/core/Tile/Tile.tsx

```diff
@@ -147,239 +105,463 @@ type TileProps = {
+    icon = <SvgNew aria-hidden />;
   }
-  if (StatusIcon) {
-    return <StatusIcon className='iui-tile-status-icon' />;
-  }
-  return null;
+
+  return children || icon ? (
+    <Box className='iui-tile-status-icon' ref={ref} {...rest}>
+      {children ?? icon}
+    </Box>
+  ) : null;
+}) as PolymorphicForwardRefComponent<'div'>;
+TileNameIcon.displayName = 'Tile.NameIcon';
+
+// ----------------------------------------------------------------------------
+// Tile.NameLabel component
+
+const TileNameLabel = React.forwardRef((props, ref) => {
+  const { children, ...rest } = props;
+  const { isActionable, isDisabled, onClick } = useSafeContext(TileContext);
+
+  return (
+    <Box as='span' className='iui-tile-name-label' ref={ref} {...rest}>
+      {isActionable && onClick ? (
+        <LinkAction
+          as='button'
+          onClick={!isDisabled ? onClick : undefined}
+          aria-disabled={isDisabled}
+        >
+          {children}
+        </LinkAction>
+      ) : (
+        children
+      )}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div'>;
+TileNameLabel.displayName = 'Tile.NameLabel';
+
+// ----------------------------------------------------------------------------
+// Tile.ContentArea component
+
+const TileContentArea = polymorphic.div('iui-tile-content');
+TileContentArea.displayName = 'Tile.ContentArea';
+
+// ----------------------------------------------------------------------------
+// Tile.Description component
+
+const TileDescription = polymorphic.div('iui-tile-description');
+TileDescription.displayName = 'Tile.Description';
+
+// ----------------------------------------------------------------------------
+// Tile.Metadata component
+
+const TileMetadata = polymorphic.div('iui-tile-metadata');
+TileMetadata.displayName = 'Tile.Metadata';
+
+// ----------------------------------------------------------------------------
+// Tile.MoreOptions component
+type TileMoreOptionsOwnProps = {
+  onButtonClick?: React.MouseEventHandler<HTMLElement>;
+  children?: React.ReactElement[];
 };
 
+const TileMoreOptions = React.forwardRef((props, ref) => {
+  const { className, children = [], onButtonClick, ...rest } = props;
+  const [isMenuVisible, setIsMenuVisible] = React.useState(false);
+
+  return (
+    <DropdownMenu
+      onShow={React.useCallback(() => setIsMenuVisible(true), [])}
+      onHide={React.useCallback(() => setIsMenuVisible(false), [])}
+      menuItems={(close) =>
+        children?.map((option: React.ReactElement) =>
+          React.cloneElement(option, {
+            onClick: (value: unknown) => {
+              close();
+              option.props.onClick?.(value);
+            },
+          }),
+        )
+      }
+    >
+      <Box
+        className={cx(
+          'iui-tile-more-options',
+          {
+            'iui-visible': isMenuVisible,
+          },
+          className,
+        )}
+        ref={ref}
+        {...rest}
+      >
+        <IconButton
+          as='button'
```

Comment:
This `as` statement can be removed since the default type for `IconButton` is a `button`

Reply:
Thanks! removed